src/test/scala-2/bincompat/ShadedGeneratedSmithySpec.scala

```diff
@@ -0,0 +1,21 @@
+package bincompat
+
+import munit.*
+
+class ShadedGeneratedSmithySpec extends FunSuite {
+  test("we don't have access to the generated KMS objects in their original package") {
+    assertNoDiff(compileErrors("""com.amazonaws.kms.KMS.id == smithy4s.ShapeId("com.amazonaws.kms", "TrentService")"""),
+      """error: object amazonaws is not a member of package com
+        |com.amazonaws.kms.KMS.id == smithy4s.ShapeId("com.amazonaws.kms", "TrentService")
+        |    ^
+        |""".stripMargin)
+  }
+
+  test("we don't have access to the generated KMS objects in the shaded package") {
```

Comment:
If this test is attempted without the extra `compile` step (i.e. `sbt clean ++2.13 test`), it will fail. If the code as-modified-by-Scalafix is compiled, the test should succeed.

The test is broken up into Scala 2 and Scala 3 variants because the compiler messages are different coming out of the different compilers. This is probably a little brittle but it should work for now.